app/src/main/java/com/cornellappdev/android/eatery/model/DiningHallModel.java

```diff
@@ -42,6 +44,97 @@ public static DiningHallModel fromEatery(Context context, boolean hardcoded,
         return model;
     }
 
+    // NOTE (yanlam): should probably move this somewhere else, but not sure where.
+    /** compareTimes(t1, t2) returns:
+     *      -1 if time of t1 comes before t2
+     *      0 if t1 is equal to t2
+     *      1 if t1 comes after t2
+     */
+    public static int compareTimes(LocalDateTime t1, LocalDateTime t2) {
+        boolean hoursEqual = t1.getHour() == t2.getHour();
+        boolean minutesEqual = t1.getMinute() == t2.getMinute();
+        if (t1.getHour() < t2.getHour()
+                || (hoursEqual && t1.getMinute() < t2.getMinute())
+                || (hoursEqual && minutesEqual && t1.getSecond() < t2.getSecond())) {
+            return -1;
+        }
+        if (t1.getHour() > t2.getHour()
+                || (hoursEqual && t1.getMinute() > t2.getMinute())
+                || (hoursEqual && minutesEqual &&t1.getSecond() > t2.getSecond())) {
+            return 1;
+        }
+        return 0;
+    }
+
+    /**
+     * getMealIntervals() returns a HashMap of all possible MealTypes and the corresponding
+     * time intervals.
+     * */
+    private HashMap<MealType, Interval> getMealIntervals() {
+        HashMap<MealType, Interval> mealIntervalMap = new HashMap<MealType, Interval>();
+        for (MealModel mealModel: getCurrentDayMenu().getAllMeals()) {
+            Interval i = mealModel.getInterval();
+            mealIntervalMap.put(mealModel.getType(), i);
+        }
+        return mealIntervalMap;
+    }
+
+    /**
+     * getCurrentMealTypeTabIndex() returns the menu tab index to be displayed, based on the
+     * current time. Note that this does NOT represent the index of the corresponding MealType enum.
+     * */
+    public int getCurrentMealTypeTabIndex() {
```

Comment:
i feel like there's a better way to calculate the meal tab index without having to loop through all the meals in `mealIntervalKeys` and use a switch statement. imo, there's a lot of redundant code in the `switch` statement and the logic can probably be condensed. 

it might be better to write a sort function or use a `PriorityQueue` for `mealIntervalKeys`, and just iterate through the pq until you find the right interval.